compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -1852,10 +1852,55 @@ impl<'a> Parser<'a> {
         let (fields, _trailing, _recovered) = self.parse_seq_to_before_end(
             &TokenKind::CloseDelim(Delimiter::Parenthesis),
             seq_sep,
-            Parser::parse_field_name,
+            |this| {
```

Comment:
I'd probably factor out this logic to a separate function and put it next to `fn parse_expr_tuple_field_access_float`.